### ISSUE TITLE
Password reset

### ISSUE THREAD

**Issue by Sjlver** [veganaut/veganaut#55]
People forget their passwords... we should email them a password reset link in this case.

This is a bit tricky because we haven't implemented emails yet. So there's some work here.

Password reset tokens are sensitive data, and should ideally be stored the same way as passwords (bcrypted, etc.)

**Comment by phoppen** [comment]
I renamed the requests to /person/isValidToken/:token and  /person/reset
As Jonas suggested, I save only a hash of the tokens now in the database. However, not using bcrypt, as bcrypt uses a salt that is individual per token, which makes it difficult to find the corresponding user in the database having only the unhashed token (this would require looking at all users and calling bcrypt.compare on these). As tokens are practically unique salt adds no security anyway, so I just used the crypto module for hashing.
